vespalib/src/vespa/vespalib/net/tls/impl/openssl_crypto_codec_impl.cpp

```diff
@@ -230,6 +234,59 @@ OpenSslCryptoCodecImpl::OpenSslCryptoCodecImpl(std::shared_ptr<OpenSslTlsContext
 
 OpenSslCryptoCodecImpl::~OpenSslCryptoCodecImpl() = default;
 
+std::unique_ptr<OpenSslCryptoCodecImpl>
+OpenSslCryptoCodecImpl::make_client_codec(std::shared_ptr<OpenSslTlsContextImpl> ctx,
+                                          const SocketSpec& peer_spec,
+                                          const SocketAddress& peer_address)
+{
+    // Naked new due to private ctor
+    return std::unique_ptr<OpenSslCryptoCodecImpl>(
+            new OpenSslCryptoCodecImpl(std::move(ctx), peer_spec, peer_address, Mode::Client));
+}
+std::unique_ptr<OpenSslCryptoCodecImpl>
+OpenSslCryptoCodecImpl::make_server_codec(std::shared_ptr<OpenSslTlsContextImpl> ctx,
+                                          const SocketAddress& peer_address)
+{
+    // Naked new due to private ctor
+    return std::unique_ptr<OpenSslCryptoCodecImpl>(
+            new OpenSslCryptoCodecImpl(std::move(ctx), SocketSpec::invalid, peer_address, Mode::Server));
+}
+
+void OpenSslCryptoCodecImpl::enable_hostname_validation_if_requested() {
+    if (_peer_spec.valid() && !_ctx->transport_security_options().disable_hostname_validation()) {
```

Comment:
should giving an invalid peer spec fail the handshake if hostname validation is enabled?